src/_global_variables.scss

```diff
@@ -35,3 +49,86 @@ $active: $brand2;
 .active {
   color: $active;
 }
+
+
+// mixins
+
+@mixin transition($property:all, $duration:0.3s, $easing:ease-in-out) {
+  transition: $property $duration $easing;
+}
+
+@mixin border-radius($radius1:3px, $radius2:3px) {
+  -webkit-border-radius: $radius1 $radius2 $radius1 $radius2;
+  -moz-border-radius: $radius1 $radius2 $radius1 $radius2;
+  border-radius: $radius1 $radius2 $radius1 $radius2;
+}
+
+@mixin box-shadow($xlength:0, $ylength:2px, $size:0, $color:rgba(30,30,30,0.25)) {
+  -webkit-box-shadow: $xlength $ylength $size $color;
+  -moz-box-shadow: $xlength $ylength $size $color;
+  box-shadow: $xlength $ylength $size $color;
+}
+
+@mixin vertical-align($position: relative) {
+  position: $position;
+  top: 50%;
+  -webkit-transform: translateY(-50%);
+  -ms-transform: translateY(-50%);
+  transform: translateY(-50%);
+}
+
+@mixin uppercase {
+  text-transform: uppercase;
+  letter-spacing: 0.125em;
```

Comment:
As a rule of thumb, I only use mixins to do "mathy" things. "If it doesn't need to pass variable, it doesn't need to be a mixin."

Instead of mixin here and for some of these others (`wordwrap`, etc), could you use a class or `@extends` instead?

Reply:
I approach mixins differently, my preference for global _snippets_ that are shared across _unrelated_ classes. You're right, I could use `extend` instead, but I prefer to reserve that for _similar_ sets of classes; to literally extent things (like `btn` > `btn-red` > `btn-small`.

This is [an elaboration on my methodology](https://csswizardry.com/2014/11/when-to-use-extend-when-to-use-a-mixin/) between them. It's ultimately a preference thing rather than a strict rule, so happy to change them if you find the approach problematic.

Reply:
Color me convinced! 👍 Great article, thanks for sharing.